assets/src/hooks/useChannel.ts

```diff
@@ -90,20 +89,28 @@ export const useCheckedChannel = <T, U>({
     if (socket !== undefined && topic !== null) {
       channel = socket.channel(topic)
       channel.on(event, ({ data: data }: { data: unknown }) => {
-        if (is(data, dataStruct)) {
+        try {
+          assert(data, dataStruct)
+
           setState(parser(data))
```

Comment:
In thinking this over more, a question I keep coming back to is what would be the benefit to users to omit data in the case that a type isn't matched? It seems to me that logging failures here has a big benefit to dev for allowing us to repair any issues in code caused by incorrect assumptions, but I wonder if that can be done without causing user-facing errors.
For example, if one vehicle on the 66 had a value of `null` for `bearing` for an unexpected reason, would it be expected to users for that vehicle and all vehicles on the 66 ladder to be omitted?
This feels to me like something Product should weigh in on, if they have not already.

Reply:
All good questions.

Re: failing to parse the whole bundle of vehicles rather than just omitting the specific ones with invalid data, I think that's an interesting idea but I'm not sure exactly how to best implement it, without specialized code that's aware for each kind of data being fetched it's an array or not. Maybe SuperStruct allows for some sort of introspection so that `useCheckedChannel` and others could see if the provided datatype is an array and if so handle it differently? But I do worry a bit about the implications of showing incomplete data vs. no data - always a complicated question.

As for failing versus logging but still passing the data on, I have mixed feelings. My worry is that if the data fails to match the type, it could be that it would have unintended behavioral consequences if passed trough to the rest of the application. For the most part these are probably harmless quirks but it's always possible that it could cause a bigger bug, and I prefer the "fail fast" approach of strong static typing. That said, it's not the way that a lot of this code was written originally, so there may be kinks to work out.

One thought: for `useVehicles` at least, maybe for now we could do the typechecking and passing errors to Sentry, but continue returning the data? Then after say a couple of weeks with no errors, we go ahead and make the change so to the functionality as implemented in this PR currently. What do you think of that?

Reply:
Yeah, I agree that checking in with product and design is a good idea. It's kind of late in the day but I'll leave myself a note to do that tomorrow. Also I'll push up a commit for the change to log to Sentry without actually erroring out for the time being.

Reply:
73d2cd00